lib/assetreqs/assetreqs.go

```diff
@@ -55,7 +67,53 @@ func New(token string, assetName string, orderNum string, cadenceName string, ca
 		fPath: filePath,
 		fName: fileName,
 		oFmt:  outputFormat,
+	}, nil
+}
+
+func ordNumIsValidFmt(orderNum string) (v bool, err error) {
+	// Note: because of no look-ahead operator (?) in go regexp, this takes multiple evaluations.
+	// Make sure that the given order number has at least one number and at least one letter, contains only digits and
+	// letters, and is 6 characters long.
+
+	// Test length = 6.
+	if len(orderNum) != 6 {
+		return false, nil
+	}
+
+	// Length is correct - make sure it doesn't contain something other than letters or digits.
+	invalid, err := hasNonLetterOrNonDigit(orderNum)
+	if err != nil {
+		return false, err
+	}
+	if invalid {
+		// contains something other than letters and digits - invalid
+		return false, err
 	}
+
+	// Make sure it contains at least one letter.
+	v, err = hasLetter(orderNum)
+	if err != nil {
+		return false, err
+	}
+	if !v {
+		// doesn't contain a letter
+		return false, err
+	}
+
+	// Contains at least one letter - make sure it contains at least one digit.
+	return hasDigit(orderNum)
```

Comment:
Similar comment as above. I think in theory an order can contain all letters (like BBBBBB), though it hasn't happened yet to my knowledge.

Reply:
for the foreseeable future, viya 4 orders will always contain both a letter and a number.